relayer/src/chain/cosmos.rs

```diff
@@ -449,10 +452,15 @@ impl CosmosSdkChain {
     /// The actual gas cost, when a transaction is executed, may be slightly higher than the
     /// one returned by the simulation.
     fn apply_adjustment_to_gas(&self, gas_amount: u64) -> u64 {
-        min(
-            gas_amount + mul_ceil(gas_amount, self.gas_adjustment()),
-            self.max_gas(),
-        )
+        debug_assert!(self.gas_adjustment() <= 1.0);
```

Comment:
I am thinking we should perhaps make this a standard `assert!` since if this invariant does not hold then it means we have a logic issue in config validation, which we probably want to hear about in production. What do you think?

Reply:
My approach to `debug_assert` is to check violations that will not break the functionality, but can be a tripwire for finding bugs in development.